packages/hub/node-tests/routes/job-tickets-test.ts

```diff
@@ -101,3 +103,146 @@ describe('GET /api/job-tickets/:id', function () {
       .expect('Content-Type', 'application/vnd.api+json');
   });
 });
+
+describe('POST /api/job-tickets/:id/retry', function () {
+  let { getJobIdentifiers, getJobPayloads, getJobSpecs } = setupStubWorkerClient(this);
+
+  this.beforeEach(function () {
+    registry(this).register('authentication-utils', StubAuthenticationUtils);
+  });
+
+  let { request, getContainer } = setupHub(this);
+  let jobTicketsQueries: JobTicketsQueries,
+    jobTicketId: string,
+    otherOwnerJobTicketId: string,
+    notFailedJobTicketId: string;
+  let ticketIds: string[];
+
+  this.beforeEach(async function () {
+    jobTicketsQueries = await getContainer().lookup('job-tickets', { type: 'query' });
+
+    jobTicketId = shortUUID.uuid();
+    await jobTicketsQueries.insert({
+      id: jobTicketId,
+      jobType: 'a-job',
+      ownerAddress: stubUserAddress,
+      payload: { 'a-payload': 'yes' },
+      spec: { 'a-spec': 'yes' },
+    });
+    await jobTicketsQueries.update(jobTicketId, { 'a-result': 'yes' }, 'failed');
+
+    otherOwnerJobTicketId = shortUUID.uuid();
+    await jobTicketsQueries.insert({ id: otherOwnerJobTicketId, jobType: 'a-job', ownerAddress: '0x111' });
+
+    notFailedJobTicketId = shortUUID.uuid();
+    await jobTicketsQueries.insert({ id: notFailedJobTicketId, jobType: 'a-job', ownerAddress: stubUserAddress });
+
+    ticketIds = [jobTicketId, otherOwnerJobTicketId, notFailedJobTicketId];
+  });
+
+  it('adds a new job, adds a job ticket for it, and returns its details', async function () {
+    await request()
+      .post(`/api/job-tickets/${jobTicketId}/retry`)
+      .set('Authorization', 'Bearer abc123--def456--ghi789')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(201)
+      .expect(function (res) {
+        expect(res.body.data.id).to.not.equal(jobTicketId);
+        expect(res.body.data.attributes).to.deep.equal({
+          state: 'pending',
+        });
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+
+    expect(getJobIdentifiers()).to.deep.equal(['a-job']);
+    expect(getJobPayloads()).to.deep.equal([{ 'a-payload': 'yes' }]);
+    expect(getJobSpecs()).to.deep.equal([{ 'a-spec': 'yes' }]);
+
+    let allTickets = await jobTicketsQueries.findAll();
```

Comment:
If we assign the return value of `await request()...` to a variable, we can retrieve the id via `res.body.data.id` and use `jobTicketQueries.find(id)` so we don't have to introduce `findAll` as a test-only method.

Reply:
👍🏻 done in a757345